repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Authorization.scala

```diff
@@ -71,83 +85,64 @@ object KeystoneV2Authorization extends LazyLogging {
         case _ =>
           logger.error("An unexpected tenant extraction type was encountered")
           None
-      }).find(_.nonEmpty).flatten
-    ).getOrElse(throw UnparsableTenantException("Could not parse tenant from the URI and/or the configured header"))
+      })
```

Comment:
Mixing lists and optionals is pretty gross

Reply:
Hah. Yes, I was conflicted about that one. It works because they are both `TraversableOnce`. I actually think an `Option` is a better type for what will only ever be zero or one values, but then it isn't strongly consistent with the collection of zero to many values I need for the header. The thing is, the semantics differ depending on what type of extraction we are performing. 

I can totally switch the `Option` to a `List` if you prefer the consistency.

Reply:
do the semantics differ? they could be doing multiple capture groups now

Reply:
Oh, are we supporting multiple capture groups in a single regex now? I was not aware of that, and did not add support for it. If that's the case, then sure, I'll add support and use a `List` here.

Reply:
I don't see why we shouldn't? It's entirely possible that the multiple tenants we get will all be in the url.

Reply:
No, that's totally legit. Since any group can be a non-capturing group, and since we are now supporting multiple tenants, there is no reason not to treat each group as a tenant.

I suppose I just didn't remember talking about that or capturing it in the story, and I didn't really think about it when I was writing this code that kept some of the old behavior despite the fact that we are changing our approach.

I will add this.